docs/01-app/03-api-reference/01-directives/use-cache-remote.mdx

```diff
@@ -0,0 +1,481 @@
+
+> **Note:** Regular `use cache` will not cache anything when used in a dynamic context (after `await connection()`, `await cookies()`, `await headers()`, etc.). Use `'use cache: remote'` to enable runtime caching in these scenarios.
+
+## Difference from `use cache` and `use cache: private`
+
+Next.js provides three caching directives, each designed for different use cases:
+
+| Feature                         | `use cache`                         | `'use cache: remote'`                                     | `'use cache: private'`              |
+| ------------------------------- | ----------------------------------- | --------------------------------------------------------- | ----------------------------------- |
+| **Works in dynamic context**    | No (requires static context)        | Yes (designed for dynamic contexts)                       | Yes                                 |
+| **Access to `await cookies()`** | No                                  | Yes (but not recommended - use `private` instead)         | Yes                                 |
+| **Access to `await headers()`** | No                                  | Yes (but not recommended - use `private` instead)         | Yes                                 |
+| **After `await connection()`**  | No (won't cache)                    | Yes (will cache)                                          | No                                  |
+| **Stored in cache handler**     | Yes (server-side)                   | Yes (server-side)                                         | No (client-side only)               |
+| **Cache scope**                 | Global (shared)                     | Global (shared)                                           | Per-user (isolated)                 |
+| **Runtime prefetchable**        | N/A (prerendered at build)          | No                                                        | Yes (when configured)               |
+| **Use case**                    | Static, shared content (build-time) | Dynamic, shared content in runtime contexts (per-request) | Personalized, user-specific content |
+
+### When to use each
+
+- **`use cache`**: Use for static content that can be prerendered at build time and doesn't depend on request-specific data.
+- **`'use cache: remote'`**: Use when caching is needed after dynamic data access (`connection()`) or for per-request caching of shared data.
+- **`'use cache: private'`**: Use for user-specific content that needs to be runtime prefetched and depends on cookies, headers, or search params.
+
+## How it works
+
+### Dynamic context detection
+
+When Next.js encounters certain APIs like `connection()`, `cookies()`, or `headers()`, the context becomes "dynamic". In a dynamic context:
+
+1. **Regular `use cache` stops working** - it won't cache anything
+2. **`'use cache: remote'` continues to work** - it caches in the remote cache handler
+3. **Results are stored server-side** in cache handlers like Redis or Vercel Data Cache
```

Comment:
this makes it sound like Next.js has some special logic for Vercel, which i think it doesn’t? i would suggest to omit any references to Vercel from API documentation pages and to use more generic language if these are expected to work with any service provider that implements the build adapter